packages/edit-post/src/components/sidebar/settings-sidebar/index.js

```diff
@@ -50,17 +56,44 @@ const SidebarContent = ( {
 	keyboardShortcut,
 	isEditingTemplate,
 } ) => {
+	const tabListRef = useRef( null );
 	// Because `PluginSidebarEditPost` renders a `ComplementaryArea`, we
 	// need to forward the `Tabs` context so it can be passed through the
 	// underlying slot/fill.
 	const tabsContextValue = useContext( Tabs.Context );
 
+	// This effect addresses a race condition caused by tabbing from the last
+	// block in the editor into the settings sidebar. Without this effect, the
+	// selected tab and browser focus can become separated in an unexpected way.
```

Comment:
For extra clarity in case someone wonders what exact scenario this hook solves.

```suggestion
	// selected tab and browser focus can become separated in an unexpected way
  // (e.g the "block" tab is focused, but the "post" tab is selected).
```